src/harness/core.ts

```diff
@@ -0,0 +1,671 @@
+// NOTE: The contents of this file are all exported from the namespace 'core'. This is to
+//       support the eventual conversion of harness into a modular system.
+
+// NOTE: Some of the functions here duplicate functionality from compiler/core.ts. They have been added
+//       to reduce the number of direct dependencies on compiler and services to eventually break away
+//       from depending directly on the compiler to speed up compilation time.
+
+namespace core {
+    export function identity<T>(v: T): T { return v; }
+
+    //
+    // Comparers
+    //
+
+    export type Comparer<T> = (x: T, y: T) => number;
+    export type EqualityComparer<T> = (x: T, y: T) => boolean;
+
+    export function compareNumbers(a: number, b: number): number {
+        if (a === b) return 0;
+        if (a === undefined) return -1;
+        if (b === undefined) return +1;
+        return a < b ? -1 : +1;
+    }
+
+    export function compareStrings(a: string, b: string, ignoreCase: boolean): number {
+        return ignoreCase
+            ? compareStringsCaseInsensitive(a, b)
+            : compareStringsCaseSensitive(a, b);
+    }
+
+    // NOTE: This is a duplicate of `compareNumbers` above, but is intended to be used only with
+    //       strings to reduce polymorphism.
+    export function compareStringsCaseSensitive(a: string, b: string): number {
+        if (a === b) return 0;
+        if (a === undefined) return -1;
+        if (b === undefined) return +1;
+        return a < b ? -1 : +1;
+    }
+
+    export function compareStringsCaseInsensitive(a: string, b: string): number {

```

Comment:
Most of these compare functions look like clones of the same thing from inside the `ts` namespace from core. If having them in this namespace is useful for convenience, IMO they should be reexported, not duplicated.

Reply:
See comment at the top of the file.